src/plugin/systems.rs

```diff
@@ -537,35 +539,83 @@ pub fn step_simulation<PhysicsHooksData: 'static + WorldQuery + Send + Sync>(
 
 /// NOTE: This currently does nothing in 2D.
 #[cfg(feature = "dim2")]
-pub fn init_async_shapes() {}
+pub fn init_async_colliders() {}
 
 /// System responsible for creating `Collider` components from `AsyncCollider` components if the
-/// corresponding resource has become available.
+/// corresponding mesh has become available.
 #[cfg(feature = "dim3")]
-pub fn init_async_shapes(
+pub fn init_async_colliders(
     mut commands: Commands,
     meshes: Res<Assets<Mesh>>,
-    async_shapes: Query<(Entity, &AsyncCollider), Without<Collider>>,
+    async_colliders: Query<(Entity, &AsyncCollider)>,
 ) {
-    for (entity, async_shape) in async_shapes.iter() {
-        match async_shape {
-            AsyncCollider::Mesh(handle) => {
-                if let Some(coll) = meshes.get(handle).and_then(Collider::bevy_mesh) {
-                    commands.entity(entity).insert(coll);
+    for (entity, async_collider) in async_colliders.iter() {
+        if let Some(mesh) = meshes.get(&async_collider.handle) {
+            match Collider::from_bevy_mesh(mesh, &async_collider.shape) {
+                Some(collider) => {
+                    commands
+                        .entity(entity)
+                        .insert(collider)
+                        .remove::<AsyncCollider>();
                 }
+                None => panic!("Unable to generate collider from mesh"),
             }
-            AsyncCollider::ConvexDecomposition(handle, params) => {
-                if let Some(coll) = meshes
-                    .get(handle)
-                    .and_then(|m| Collider::bevy_mesh_convex_decomposition_with_params(m, params))
-                {
-                    commands.entity(entity).insert(coll);
+        }
+    }
+}
+
+/// System responsible for creating `Collider` components from `AsyncSceneCollider` components if the
+/// corresponding scene has become available.
+#[cfg(feature = "dim3")]
+pub fn init_async_scene_colliders(
+    mut commands: Commands,
+    meshes: Res<Assets<Mesh>>,
+    scenes: Res<Assets<Scene>>,
+    async_colliders: Query<(Entity, &AsyncSceneCollider)>,
+    children: Query<&Children>,
+    mesh_handles: Query<(&Name, &Handle<Mesh>)>,
+) {
+    for (entity, async_collider) in async_colliders.iter() {
+        if scenes.get(&async_collider.handle).is_some() {
+            for child in get_children_recursively(entity, &children) {
```

Comment:
Instead of building a `LinkedList` with all the descendants, I think it would be more efficient and versatile to replace `get_children_recursively` by a `traverse_descendants` function that would look like this:
```rust
fn traverse_descendants(entity: Entity, children: &Query<&Children>, f: &mut impl FnMut(Entity)) {
    if let Ok(entity_children) = children.get(entity) {
        for child in entity_children.iter() {
            f(child);

            let mut children = get_children_recursively(*child, children);
            all_children.append(&mut children, f);
        }
    }
}
```

So this `for child in get_children_recursively(entity, &children)` would simply call `traverse_descendants` and provide a closure.